src/CloudNative.CloudEvents.AspNetCore/HttpRequestExtension.cs

```diff
@@ -45,90 +22,54 @@ public static class HttpRequestExtension
         /// <param name="extensions">List of extension instances</param>
         /// <returns>A CloudEvent instance or 'null' if the request message doesn't hold a CloudEvent</returns>
         public static async ValueTask<CloudEvent> ReadCloudEventAsync(this HttpRequest httpRequest,
-            ICloudEventFormatter formatter = null,
-            params ICloudEventExtension[] extensions)
+            ICloudEventFormatter formatter,
+            params CloudEventAttribute[] extensionAttributes)
         {
-            if (httpRequest.ContentType != null &&
-                httpRequest.ContentType.StartsWith(CloudEvent.MediaType,
-                    StringComparison.InvariantCultureIgnoreCase))
+            if (HasCloudEventsContentType(httpRequest))
             {
-                // handle structured mode
-                if (formatter == null)
-                {
-                    // if we didn't get a formatter, pick one
-                    var contentType = httpRequest.ContentType.Split(';')[0].Trim();
-                    if (contentType.EndsWith(JsonEventFormatter.MediaTypeSuffix,
-                        StringComparison.InvariantCultureIgnoreCase))
-                    {
-                        formatter = jsonFormatter;
-                    }
-                    else
-                    {
-                        throw new InvalidOperationException("Unsupported CloudEvents encoding");
-                    }
-                }
-
-                return await formatter.DecodeStructuredEventAsync(httpRequest.Body, extensions);
+                // TODO: Handle formatter being null
+                return await formatter.DecodeStructuredEventAsync(httpRequest.Body, extensionAttributes).ConfigureAwait(false);

```

Comment:
Do we not want to support jsonFormatter as a default anymore?

Reply:
Nope - I want to move the JSON event formatter out to a separate package, so we can have one based on System.Text.Json and one based on Json.NET. See https://github.com/cloudevents/sdk-csharp/issues/42

(In general I'd like to try to limit the JSON-centricity of the codebase. The spec is already a bit overly-JSON-centric for my liking, but we can try to reduce the blast radius of that.)